src/components/Popup/Popup.tsx

```diff
@@ -146,15 +182,19 @@ export default class Popup extends AutoControlledComponent<ReactProps<PopupProps
       this.trySetOpen(!this.state.open, e, true)
     },
     closeAndFocusTrigger: e => {
-      this.closeAndFocusTrigger(e)
+      this.closeAndFocusTrigger(e, true)
+      e.stopPropagation()
+    },
+    close: e => {
+      this.closeAndFocusTrigger(e, false)
       e.stopPropagation()
     },
   }
 
-  private closeAndFocusTrigger = e => {
+  private closeAndFocusTrigger = (e, focusTrigger) => {
     if (this.state.open) {
       this.trySetOpen(false, e, true)
-      _.invoke(this.triggerDomElement, 'focus')
+      focusTrigger && _.invoke(this.triggerDomElement, 'focus')
```

Comment:
seems to be a bit confusing having name of function `closeAndFocusTrigger`, but it does what it says only in one case. What do you if to change it similar to `setState` structure?

```js
...
protected actionHandlers: AccessibilityActionHandlers = {
...
    closeAndFocusTrigger: e => {
      this.close(e, () => _.invoke(this.triggerDomElement, 'focus'))
      e.stopPropagation()
    },
    close: e => {
      this.close(e)
      e.stopPropagation()
    },
  }
...

private close = (e, onClose?: () => void) => {
    if (this.state.open) {
      this.trySetOpen(false, e, true)
      onClose && onClose()
    }
  }
```

Reply:
Will refactor it a bit.